src/daft-plan/src/display/tree.rs

```diff
@@ -0,0 +1,159 @@
+use std::{
+    fmt::{self, Write},
+    sync::Arc,
+};
+
+pub trait TreeDisplay {
```

Comment:
can we move `TreeDisplay` and `MermaidDisplayVisitor` to a common crate so I can also leverage them for the pipeline visualizer :)